common/values.yaml

```diff
@@ -1,78 +1,81 @@
-#     domain: "{{ .Values.expose.config.domain }}"
-#     http: "{{ .Values.expose.config.http }}"
-#     tlsacme: "{{ .Values.expose.config.tlsacme }}"
-expose:
-  config:
-    domain: ""
-    http: "true"
-    tlsacme: "false"
-
-ingress:
-  ## Set this to true in order to enable TLS on the ingress record
-  tls: 
-
-  ## Empty default. Each ingress should provide its own value or template
-  path: 
-
-  ## If TLS is set to true, you must declare what secret will store the key/certificate for TLS
-  tlsSecret: 
-
-  ## If set, overrides .Values.global.gateway.host configuration 
-  hostName: 
- 
-  ## Configure default ingress annotations 
-  annotations: {} 
+# common values shared between all charts
+
+global:
+  gateway:
+    ## Configure default domain for gateway host, i.e. "{{ .Relase.Name }}.127.0.0.1.nip.io"
+    domain: ""
+    ## Confige default gateway host Helm template, i.e. "activiti-cloud-gateway.{{ .Values.global.gateway.domain }}"
+    host: ""
+    ## Toggle creating http or https ingress rules, supports literal or boolean values
+    http: true
+
+    ## Used to enable automatic TLS for ingress if http is false
+    tlsacme: false
+
+  keycloak:
+    enabled: true
+    ## Overrides gateway host configuration
+    url: "" 
+    ## Configure default keycloak path 
+    path: "/auth"
+    ## Configure default keycloak host template, i.e "activiti-keycloak.{{ .Values.global.gateway.domain }}"
+    host: ""
+
+    ## Configure default Keycloak realm
+    realm: "activiti"
+    resource: "activiti"
+    client: "activiti"
+
+    ## Adds Keycloak extraEnv to deployments
+    extraEnv: ""
+
+  ## Configure pull secrets for all deployments
+  registryPullSecrets: []
+
+  ## Adds global extraEnv to deployments
+  extraEnv: ""
+
+## Adds extraEnv to deployments
+extraEnv: ""
+
+## Configures additional pull secrets for this deployment
+registryPullSecrets: []
+
+## Allows to template gateway values from Jx Expose controller configuration in GitOps environment, i.e.
+# expose:
+#   config:
+#     domain: 1.2.3.4.nip.io
+#     http: "true"
+#     tlsacme: "false"
+#
+# global:
+#   keycloak:
+#     host: "activiti-keycloak.{{ tpl .Values.global.gateway.domain . }}"
+#   gateway:
+#     domain: "{{ .Values.expose.config.domain }}"
+#     http: "{{ .Values.expose.config.http }}"
+#     tlsacme: "{{ .Values.expose.config.tlsacme }}"
+expose:
+  config:
+    domain: ""
+    http: "true"
+    tlsacme: "false"
+
+ingress:
+  ## Set this to true in order to enable TLS on the ingress record
+  tls: 
+
+  ## Empty default. Each ingress should provide its own value or template
+  path: 
+
+  ## If TLS is set to true, you must declare what secret will store the key/certificate for TLS
+  tlsSecret: 
+
+  ## If set, overrides .Values.global.gateway.host configuration 
+  hostName:
+ 
+  ## Configure default ingress annotations 
+  annotations: {}
+
+  ## If set to true, disable default hostname to .Values.global.gateway.host
+  skipDefaultHostName:
```

Comment:
@mteodori what the ...? what does this mean? Why would we do this of not having a host name?? Can you please explain?

Reply:
It’s an option I need to pull everything together while there are some charts that still don’t have the hostname as one single hostname rule kills any non hostname one can you please merge and publish thanks

Reply:
@mteodori I do not really understand this.. and I check with @igdianov he is in the same situation.. we can just not merge stuff if we don't understand or we can test this. It still doesn't make sense to me.. we can catch up on this on monday

Reply:
this is an option to avoid forcing the hostname to the gateway one on ingress, which breaks any existing ingress of other charts installed that don't explicitly set it, as I wrote in my comment before, @igdianov is aware of it